### PR TITLE
support friction CCFLOAT and fix web ligatures' width

### PR DESCRIPTION
**Pull request by longchuan**
Re: #3391

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/physics/framework/assets/physics-material.ts [hunk 1]
```diff
@@ -66,10 +66,8 @@ export class PhysicsMaterial extends Asset {
     }
 
     set friction (value) {
-        if (!math.equals(this._friction, value)) {
-            this._friction = value;
-            this.emit(PhysicsMaterial.EVENT_UPDATE);
-        }
+        this._friction = value;
+        this.emit(PhysicsMaterial.EVENT_UPDATE);
```
Comment (minggo) [inline cocos/physics/framework/assets/physics-material.ts:70]: 两个浮点数如果太接近，不修改有什么影响呢？

**Comment by minggo** [comment]
另外，这个 PR 处理了两件事情，麻烦分开提交。谢谢。

File: cocos/physics/framework/assets/physics-material.ts (hunk 1 above)
Comment (longchuan) [reply cocos/physics/framework/assets/physics-material.ts:70]: > 两个浮点数如果太接近，不修改有什么影响呢？

在不随机的情况下，结果会一致。热启动解算器默认随机，所以还要增加重置接口

**Comment by longchuan** [comment]
> 另外，这个 PR 处理了两件事情，麻烦分开提交。谢谢。

不太会用这个，还在熟悉

**Pull request closed without merging**